reporter/src/main/kotlin/reporters/NoticeTemplateReporter.kt

```diff
@@ -19,15 +19,202 @@
+            tagSyntax = Configuration.SQUARE_BRACKET_TAG_SYNTAX
+            templateExceptionHandler = TemplateExceptionHandler.RETHROW_HANDLER
+            templateLoader = ClassTemplateLoader(this@NoticeTemplateReporter.javaClass.classLoader, "template.notice")
+            wrapUncheckedExceptions = true
+        }
+
+        val templateIds = options[OPTION_TEMPLATE_ID]?.split(",") ?: listOf("default")
+        val templatePaths = options[OPTION_TEMPLATE_PATH]?.split(",").orEmpty()
+
+        val templateFiles = templatePaths.map { path ->
+            File(path).expandTilde().also {
+                require(it.exists()) { "Could not find template file at ${it.absolutePath}." }
+            }
+        }
+
+        val outputFiles = mutableListOf<File>()
+
+        templateIds.forEach { id ->
+            log.info { "Generating notice file using template id '$id'." }
+
+            val outputFile = outputDir.resolve("$NOTICE_FILE_PREFIX$id")
+            outputFiles += outputFile
+
+            val template = freemarkerConfig.getTemplate("$id.ftlh")
+            outputFile.writer().use { template.process(dataModel, it) }
+        }
+
+        templateFiles.forEach { file ->
+            log.info { "Generating notice file using template file '${file.absolutePath}'." }
+
+            val outputFile = outputDir.resolve("$NOTICE_FILE_PREFIX${file.nameWithoutExtension}")
+            outputFiles += outputFile
+
+            freemarkerConfig.setDirectoryForTemplateLoading(file.parentFile)
+
+            val template = freemarkerConfig.getTemplate(file.name)
+            outputFile.writer().use { template.process(dataModel, it) }
+        }
+
+        return outputFiles
+    }
+
+    /**
+     * License information for a single package or project.
+     */
+    class PackageNoticeModel(
+        val id: Identifier,
+        private val input: ReporterInput
+    ) {
+        /**
+         * True if the package is excluded.
+         */
+        val excluded: Boolean by lazy { input.ortResult.isExcluded(id) }
+
+        /**
+         * The resolved license information for the package.
+         */
+        val license: ResolvedLicenseInfo by lazy {
+            val resolved = input.licenseInfoResolver.resolveLicenseInfo(id)
+            resolved.copy(licenses = resolved.licenses.sortedBy { it.license.toString() })
+        }
+
+        /**
+         * The resolved license file information for the package.
+         */
+        val licenseFiles: ResolvedLicenseFileInfo by lazy { input.licenseInfoResolver.resolveLicenseFiles(id) }
+
+        /**
+         * Returns all [ResolvedLicense]s for this package excluding those licenses which are contained in any of the
+         * license files. This is useful when the raw texts of the license files are included in the generated notice
+         * file and all licenses not contained in those files shall be listed separately.
+         */
+        fun licensesNotInLicenseFiles(): List<ResolvedLicense> {
```

Comment:
IDE inspections seems to thing that this also needs

        @Suppress("UNUSED") // This function is used in the templates.

Reply:
Weird, it didn't highlight the function for me. Done.